### ISSUE TITLE
Bug: CountAsync (PostgreSql) fails when querying by Enum.

### ISSUE THREAD

**Issue by atresnjo** [mikependon/RepoDB#798]
### Bug Description

I'm using EF Core code first approach. Basically my goal is to use EF Core for the writes and RepoDB for the reads, so far working good, but I'm having issues with a query where I try to get total count by an enum property.

**Exception Message:**

```csharp
System.InvalidCastException: Can't write CLR type bazaarly.Domain.UserTaskStatus with handler type Int32Handler
         at lambda_method154(Closure , NpgsqlTypeHandler , Object , NpgsqlLengthCache& , NpgsqlParameter )
         at Npgsql.TypeHandling.NpgsqlSimpleTypeHandler`1.ValidateObjectAndGetLength(Object value, NpgsqlLengthCache& lengthCache, NpgsqlParameter parameter)
```

**Schema and Model:**

![image](https://user-images.githubusercontent.com/41112852/112450051-420f4b80-8d54-11eb-8e74-fd96c6238d5e.png)

```csharp
    [Map("UserTasks")]
    public class UserTask
    {
        public Guid Id { get; set; }
        public List<ParseResult> ParseResults { get; set; } = new();
        public string SearchTerm { get; set; }
        public DateTime CreatedDate { get; set; }
        public UserAccount UserAccount { get; set; }
        public Guid UserAccountId { get; set; }
        public SiteType SiteType { get; set; }
        public DateTime LastRequest { get; set; }
        public UserTaskStatus Status { get; set; }
        public int ProductsExtractedCount { get; set; }
        public UserTask() => CreatedDate = DateTime.UtcNow;
    }
```

Code:

```csharp
var activeTasksRepo = await connection.CountAsync<Domain.UserTask>(x => x.Status == UserTaskStatus.Active, cancellationToken: cancellationToken);

var activeTasksEF = await _databaseContext.UserTasks.CountAsync(x => x.Status == UserTaskStatus.Active, cancellationToken);
```

The EF call works fine, what am I doing wrong? Thanks for the great work! 👍 

**Library Version:**

    <PackageReference Include="RepoDb" Version="1.12.8-beta4" />
    <PackageReference Include="RepoDb.PostgreSql" Version="1.1.4-beta1" />

**Comment by gloriousjob** [comment]
What is the UserTaskStatus class? My guess is that translation may be trying a string form of the enum instead of an int but it's just a guess.

**Comment by atresnjo** [comment]
It's an enum. :) 

```csharp
    public enum UserTaskStatus
    {
        Active,
        Inactive,
        Successful,
    }
```

**Comment by mikependon** [comment]
Referencing #787.

**Comment by mikependon** [comment]
The fix has been issued and it will be available on the next version (> v1.1.4-beta1).
